src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Internal/Telemetry/TelemetryCommonProperties.cs

```diff
@@ -42,14 +40,14 @@ public Dictionary<string, string> GetTelemetryCommonProperties()
                 {RuntimeId, RuntimeEnvironment.GetRuntimeIdentifier()},
                 {ProductVersion, _productVersion},
                 {DockerContainer, IsDockerContainer()},
-                {MacAddressHash, GetMacAddress() },
+                {MacAddressHash, await GetMacAddressAsync() },
                 {KernelVersion, GetKernelVersion()}
             };
     }
 
-    private string GetMacAddress()
+    private async Task<string> GetMacAddressAsync()
     {
-        var macAddress = _environmentService.GetMacAddress();
+        string macAddress = await _environmentService.GetMacAddressAsync() ?? string.Empty;
         return macAddress != null ? macAddress.Hash() : Guid.NewGuid().ToString();
```

Comment:
This changes behaviour a bit. With this new code, `macAddress` can never be `null`, so the return string will the hash of the empty string rather than a new GUID.